tempesta_fw/t/unit/test_http_parser.c

```diff
@@ -1485,6 +1485,40 @@ TEST(http_parser, accept)
 	{
 		EXPECT_TRUE(test_bit(TFW_HTTP_B_ACCEPT_HTML, req->flags));
 	}
+
+	FOR_REQ("GET / HTTP/1.1\r\n"
+		"Accept:  invalid/invalid;  q=0.5;    key=val, */* \r\n"
+		"\r\n")
+	{
+		EXPECT_TRUE(test_bit(TFW_HTTP_B_ACCEPT_HTML, req->flags));
+	}
+
+	FOR_REQ("GET / HTTP/1.1\r\n"
+		"Accept:  text/html,  invalid/invalid;  key=val;   q=0.5 \r\n"
+		"\r\n")
+	{
+		EXPECT_TRUE(test_bit(TFW_HTTP_B_ACCEPT_HTML, req->flags));
+	}
+
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\r\n"
+		"Accept: */invalid\r\n"
+		"\r\n");
+
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\r\n"
+		"Accept: invalid/invalid; q=foo\r\n"
+		"\r\n");
+
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\r\n"
+		"Accept: invalid\r\n"
+		"\r\n");
+
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\r\n"
+		"Accept: /invalid\r\n"
+		"\r\n");
+
+	EXPECT_BLOCK_REQ("GET / HTTP/1.1\r\n"
+		"Accept: invalid/\r\n"
+		"\r\n");
```

Comment:
Some cases are missing and the parsed can't pass them: 
* `text/html; q=0.5`, `text/html ; q=0.5` - those with valid type-sybtype and parameter, but with different number of  OWS between them.
* `*/html` - forbidden by RFC, but allowed by parser (bug).
* `text/html; =0.5` - no token in  parameter, `text/html; q = 0.5` - OWS inside parameter
* `text/html; param="value"`  - quoted-string inside accept extension, the string may include ows and other stuff.

Please, take a look into [RFC](https://tools.ietf.org/html/rfc7231#section-5.3.2) and add all possible Accept header mutations, that can still be parsed successfully.

Reply:
Mozilla also tells about [Cross-Origin Resource Sharing safety for Accept header](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Accept);
>  additional restriction that values can't contain a CORS-unsafe request header byte: "():<>?@[\]{}, Delete, Tab and control characters: 0x00 to 0x19.

 I don't know, whether it should be taken into account, while parsing the quoted strings. Anyway, the quoted-strings is a good candidate to have attacks in.
> accept-ext = OWS ";" OWS token [ "=" ( token / quoted-string ) ]

 Attacker may use the difference between quoted string processing between Tempesta and backends and use it fot request-splitting attacks.

Reply:
In `Req_I_AcceptOther` state we use `uri` alphabet to check the input. The alphabet (see `str_avx2.S:94`) doesn't allow `DEL`, `TAB`, any control characters (`0x00-0x20`), `<`, `>`, `\`, but does allow `():?@[]`. The alphabet was choosen according to the RFC, so I'm not sure how important to exclude the rest of allowed chars.

Reply:
Now we used `token` alphabet. We parse params on `token`=`token`